WIP: Upgrade to python 3

kovalch opened the pull request.

BWibo committed:
Fix pwd validation

Add missing `@` char to pwd validation, fixes #6

blagojabozinovski committed:
Merge pull request #11 from BWibo/patch-1

Fix pwd validation

kovalch committed:
feat: Upgrade to py3 using 2to3

kovalch committed:
fix: Remove python-future that was used to make the scripts py2 compatible

kovalch committed:
fix: Update init file

kovalch committed:
fix: Install the missing webob module

kovalch committed:
fix: Remodjust import to work with either CKAN 2.8 or 2.9+

kovalch committed:
feat: Refactored the Authentication logic to use Flask-login instead of repoze.who

stefina committed:
feat: copy template from CKAN 2.11.3 to override it

stefina committed:
style: fix indentation to protect me from eyesore

stefina committed:
feat: Enable sysadmins to reset locked-out users

Re-apply the commit with the actual change of the template to enable sysadmins to reset locked-out users.

stefina committed:
feat: authenticate user through CKANs authenticator

stefina committed:
fix: forward user to dashboard when logging in

stefina committed:
chore: remove webob as we do not use it

stefina committed:
style: isort imports

stefina committed:
style: apply black style

stefina committed:
feat: update build pipeline and add simple tests

stefina committed:
style: reorder imports

stefina committed:
Merge branch 'master' into refactor-build

stefina committed:
Merge pull request #16 from opendata-swiss/refactor-build

Refactor build

bellisk commented on .github/workflows/test.yml:
```diff
@@ -1,26 +1,56 @@
+        with:
+          python-version: '3.10'
+      - name: Install requirements
+        run: pip install flake8 pycodestyle black[python2]==21.12b0 click==8.0.4 isort
```
```suggestion
        run: pip install flake8 pycodestyle black isort
```

We don't need to pin the `black` version any more because we don't want to check `python2` codestyle :grin:

stefina committed:
fix: use black on the latest version and remove click

Co-authored-by: Rae Knowler <bellisk@users.noreply.github.com>

bellisk committed:
feat: Override new snippet from CKAN core

bellisk committed:
feat: Display user login status to sysadmins

bellisk committed:
feat: Update to match latest template in CKAN core

bellisk committed:
feat: Remove out-of-date template

bellisk committed:
feat: Update template with changes from CKAN core

bellisk committed:
tests: Update path to test-core.ini in test.ini

bellisk committed:
Merge pull request #17 from opendata-swiss/feat/small-updates

Feat/small updates

bellisk committed:
feat: Don't use six now that we only support Python 3

Also, don't use 'str' as a validator for the password2 field, as CKAN no longer allows builtin functions as validators.

bellisk committed:
feat: Update setup.py with supported python versions

bellisk committed:
Merge pull request #18 from opendata-swiss/feat/get-rid-of-six

Feat/get rid of six

bellisk committed:
fix: Base our user edit schema on the CKAN user_edit_form_schema

The CKAN user_edit_form_schema is the one actually used when validating the data passed in when we want to update a user from the web UI. We need to base our custom schema on it.

bellisk committed:
feat: Update our custom _prepare methods based on CKAN 2.11.3

We should base our approach on the one used in CKAN core as much as possible to avoid surprises that are hard to debug.

bellisk committed:
style: Don't split string into two and then re-merge it

bellisk committed:
Merge pull request #19 from opendata-swiss/fix/use-correct-schema-for-editing-user

Fix/use correct schema for editing user

bellisk committed:
chore: Clean up unused code

The config options 'app' and 'flask_app' are never defined, so this code was never called or used.

bellisk committed:
Merge pull request #20 from opendata-swiss/chore/clean-up-unused-code

chore: Clean up unused code

bellisk committed:
feat: Declare config options in config_declaration.yaml

bellisk committed:
Merge pull request #21 from opendata-swiss/feat/declare-config-options

feat: Declare config options in config_declaration.yaml

bellisk committed:
fix: Update form input error arguments on perform_reset.html template

This fixes an error when the user clicks the password-reset link.

bellisk committed:
Merge pull request #22 from opendata-swiss/fix/fix-error-on-password-reset-page

fix: Update form input error arguments on perform_reset.html template

bellisk committed:
feat: Update to pyproject.toml packaging

See https://github.com/pypa/pip/issues/11457 and https://github.com/ckan/ckan/issues/8382.

bellisk committed:
tests: Install ckanext in modern way

bellisk committed:
tests: Don't apply pending migrations again

They are applied in db init.

bellisk committed:
Update CKAN 2.11 compatibility status to 'Yes'

bellisk committed:
Merge pull request #23 from opendata-swiss/feat/transition-to-pyproject.toml

feat: Update to pyproject.toml packaging
